capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ExtractionsActivity.kt

```diff
@@ -17,34 +17,41 @@ import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.google.android.material.textfield.TextInputEditText
 import com.google.android.material.textfield.TextInputLayout
+import dagger.hilt.android.AndroidEntryPoint
 import net.gini.android.capture.Amount
 import net.gini.android.capture.AmountCurrency
 import net.gini.android.capture.GiniCapture
 import net.gini.android.capture.network.GiniCaptureDefaultNetworkService
+import net.gini.android.capture.network.GiniCaptureNetworkService
 import net.gini.android.capture.network.model.GiniCaptureCompoundExtraction
 import net.gini.android.capture.network.model.GiniCaptureSpecificExtraction
+import net.gini.android.capture.screen.R
 import net.gini.android.capture.screen.databinding.ActivityExtractionsBinding
 import net.gini.android.core.api.models.SpecificExtraction
 import org.slf4j.LoggerFactory
 import java.io.File
 import java.io.FileOutputStream
 import java.io.IOException
 import java.math.BigDecimal
-import java.util.*
+import java.util.Date
+import javax.inject.Inject
 
 /**
  * Displays the Pay5 extractions: paymentRecipient, iban, bic, amount and paymentReference.
  *
  * A menu item is added to send feedback.
  */
-class ExtractionsActivity : AppCompatActivity(), ExtractionsAdapterImpl.ExtractionsAdapterInterface {
+
+@AndroidEntryPoint
+class ExtractionsActivity : AppCompatActivity(),
+    ExtractionsAdapterImpl.ExtractionsAdapterInterface {
     private lateinit var binding: ActivityExtractionsBinding
 
     private var mExtractions: MutableMap<String, GiniCaptureSpecificExtraction> = HashMap()
     private var mCompoundExtractions: Map<String, GiniCaptureCompoundExtraction> = HashMap()
     private val mLegacyExtractions: MutableMap<String, SpecificExtraction> = HashMap()
     private var mExtractionsAdapter: ExtractionsAdapter<Any>? = null
-    private lateinit var defaultNetworkService: GiniCaptureDefaultNetworkService
+    @Inject lateinit var giniCaptureNetworkService: GiniCaptureNetworkService
```

Comment:
[Suggestion] Is it possible to have the type be `GiniCaptureDefaultNetworkService` while the injection provider uses `GiniCaptureNetworkService`? This would avoid us having to cast to `GiniCaptureDefaultNetworkService` on line 82.